numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1788,6 +1791,24 @@ unpack_bits(PyObject *input, int axis, PyObject *count_obj, char order)
         goto fail;
     }
 
+    /*
+     * setup lookup table under GIL, 256 8 byte blocks representing 8 bits
+     * expanded to 1/0 bytes
+     */
+    if (unpack_init == 0) {
+        npy_uint64 j;
+        for (j=0; j < 256; j++) {
+            npy_uint64 k;
+            for (k=0; k < 8; k++) {
+                npy_uint8 v = (j & (1 << k)) == (1 << k);
+                unpack_lookup_b[j][7 - k] = v;
```

Comment:
Is it possible to detect endian-ness and initialize just one table, or do we need both on a given platform?

Reply:
Also, is it possible to change the access rather than changing the table? Might be a bit more maintainable.

Reply:
No, we need both to support the two modes of unpack bits. Platform endianness is irrelevant.

Reply:
Creating two tables is done so both can be memcopy'd

Reply:
one could byteswap from one table, but that is about 60% slower than a second table.
two extra kb of memory usage should not really matter for typical numpy applications though as the function is not that important and the slowdown would be in the new option only it could be considered

Reply:
updated to do that, a bit slower but probably fine

Reply:
The comment about changing access still holds.